MapboxNavigation/NavigationViewController.swift

```diff
@@ -369,6 +371,10 @@ public class NavigationViewController: NavigationPulleyViewController, RouteMapV
             }
         }
         
+        if route != routeController.routeProgress.route {
```

Comment:
If we don't override isEqual, it ends up doing a pointer equality check. That won't be as thorough.